src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -23,10 +26,69 @@
 
     computed: {
       ...mapState('flow', ['flows']),
+
+      canvasHeight() {
+        const defaultHeight = window.screen.availHeight
+        if (!this.flows[0].blocks && this.flows[0].blocks.length) {
+          return defaultHeight
+        }
+
+        const blockAtTheLowestPosition = lodash.maxBy(this.flows[0].blocks, 'platform_metadata.io_viamo.uiData.yPosition')
+
+        if (!blockAtTheLowestPosition) {
+          return defaultHeight
+        }
+
+        const blockElement = document.getElementById(`block/${blockAtTheLowestPosition.uuid}`)
+
+        if (!blockElement) {
+          return defaultHeight
+        }
+
+        const blockHeight = blockElement.clientHeight
+        return lodash.get(blockAtTheLowestPosition, 'platform_metadata.io_viamo.uiData.yPosition') + blockHeight + 100
+      },
+
+      canvasWidth() {
+        const defaultWidth = window.screen.availWidth
+        if (!this.flows[0].blocks && this.flows[0].blocks.length) {
+          return defaultWidth
+        }
+
+        const blockAtTheFurthestRightPosition = lodash.maxBy(this.flows[0].blocks, 'platform_metadata.io_viamo.uiData.xPosition')
+
+        if (!blockAtTheFurthestRightPosition) {
+          return defaultWidth
+        }
+
+        const blockElement = document.getElementById(`block/${blockAtTheFurthestRightPosition.uuid}`)
+
+        if (!blockElement) {
+          return defaultWidth
+        }
+
+        const blockWidth = blockElement.clientWidth
+        return lodash.get(blockAtTheFurthestRightPosition, 'platform_metadata.io_viamo.uiData.xPosition') + blockWidth + 100
```

Comment:
I think it could be important for us to separate these methods into a few parts to allow vuejs to more efficiently trigger recache of these as computed properties. A couple things I'm thinking about: 
1. Are we mixing domain data lookup and UI manipulation (eg. dom access, window dimensions, etc)
2. How frequently is the underlying data that we're accessing going to change? 

A suggestion:
Create separate computer getters for things returned from dom access and those that specifically do analysis over the underlying data (eg. maxBy):
- `largestXPositionedBlockId()`
- `largestXPositionedBlock()` — Set up refs on block and index into that locally via `$refs[largestXPositionedBlockId()]`; this could prevent having to call `getElementById()` at all. 
- `windowWidth()` — I'm actually wholly uncertain as to how vuejs can trigger on changes to window properties, keep an eye out for this as a place where changes and redraws may not be propagating as anticipated. An alternative could be to observe window's resize event and set that explicitly so that we've got that cached locally.

Any accesses to the DOM should be considered equivalent to us going to the DB on the server side, fast, but very slow compared to having things local or cached. They're often fast enough, but in situations like this where we're dispatching many events per second while a person is dragging, and doing work to repaint connections, we need to pay attention to efficiency. 

Let me know if you'd like to chat this one out! 😸

Reply:
@bzabos , I think I need a catchup about `$refs[largestXPositionedBlockId()]` & `windowWidth()`

Reply:
Hi @safydy, here are the resources we'd discussed: https://developers.google.com/web/fundamentals/performance/rendering